### PR TITLE
Fix small typo, add non-JS version

### PR DESCRIPTION
**Pull request by kescherCode**
I've gone ahead and fixed a small typo in CheckFormat.js (validility -> validity), and added the entirety of the page, minus the things that really do require JavaScript, to the index.html. This allows non-JS clients to view the page as well (tested in Firefox with NoScript, as well as `elinks`).

### CODE REVIEW

**Comment by koutsie** [comment]
Huge +1

File: public/index.html [hunk 1]
```diff
@@ -1,20 +1,18 @@
...
-    <link rel="icon" href="%PUBLIC_URL%/favicon.ico" />
-    <meta name="viewport" content="width=device-width, initial-scale=1" />
-    <meta name="theme-color" content="#000000" />
-    <meta
-      name="description"
-      content="Comparison between RFC 3339 and ISO 8601 date formats"
-    />
-    <link rel="apple-touch-icon" href="%PUBLIC_URL%/logo192.png" />
+<head>
+    <meta http-equiv="content-type" content="text/html; charset=UTF-8">
```
Comment (IJMacD) [inline public/index.html:4]: L4 - This line is not necessary

**Review by IJMacD — COMMENTED** [review]
Thanks for catching the typo.

Do you have a script to generate the non-JS version to make updates easier when formats may be added/removed in the future?

File: public/index.html (hunk 1 above)
Comment (kescherCode) [reply public/index.html:4]: This line is in the JavaScript version when run with JavaScript.

Comment (kescherCode) [reply public/index.html:4]: But sure, I can remove it.

**Comment by kescherCode** [comment]
@IJMacD I have added a getStaticSite.js, which is intended to be run in browser DevTools, for example while doing `yarn start`. The downloaded file can be placed at /public/index.html. (That parser might need some adjustment in the future, however.)

**Comment by koutsie** [comment]
> @IJMacD I have added `getStaticSite.js`, which is intended to be run in browser DevTools, for example while doing `yarn start`. The downloaded file can be placed at /public/index.html. (That parser might need some adjustment in the future, however.)

Tested and seems to work!

**Review by koutsie — APPROVED** [review]

**Comment by IJMacD** [comment]
I put the style back in a `noscript` tag as you had originally to avoid interference with live styles if available (in dev or prod).

I've tried running some performance comparisons to see if adding 80 kB for all users for the benefit of a tiny minority of users causes noticeable issues. On my laptop with a good connection I see no problems so we're all good there. 👍Still haven't checked on mobile/slow connection though.

||Resources|Gzip'd|
|-|--------------|------------|
|Before|190 kB|54.8 kB|
|After|271 kB|64.1 kB|
|Delta|+43%|+17%|

I had the thought of linking to the GitHub readme in the `noscript` tag for those who elect to run without JS. That would avoid the problem of bloated page loads for little noticeable benefit. The `getStaticSite.js` could be repurposed to update README.md instead.

How does this sound?

**Comment by kescherCode** [comment]
Indeed, I forgot to put the style into a noscript tag. Oops!

Now, there are some different approaches that one could take here. First of all, one could generate a separate page to begin with for JS users, with automatic redirection to the non-JS version using Http-Equiv "Refresh", and back to the JS version using JS when enabled. I used that approach at one of [my projects](https://doggos.kescher.at) (the source code is linked, too). Note that it will redirect you to /nojs when JS is disabled.
And my `getStaticSite.js` script could potentially be obsoleted by a script https://reactjs.org/docs/react-dom-server.html, as a friend pointed out, that is run with Node instead of browser DevTools.

**Comment by IJMacD** [comment]
Server side rendering has always been an option but I think in our case with the extreme size of the markup, any benefits gained from SSR are outweighed by the additional payload; both in terms of waiting for the network and additional parsing requirements.

I've changed the getStaticSite.js script into a generateReadme.js script. This means there is only one static version to maintain and is accessible to everyone. The main page will redirect non-JS users back to the static version hosted on GitHub as the README.

**Comment by IJMacD** [comment]
Server side rendering has always been an option but I think in our case with the extreme size of the markup, any benefits gained from SSR are outweighed by the additional payload; both in terms of waiting for the network and additional parsing requirements.

I've changed the getStaticSite.js script into a generateReadme.js script. This means there is only one static version to maintain and is accessible to everyone. The main page will redirect non-JS users back to the static version hosted on GitHub as the README.

**Comment by koutsie** [comment]
Works I guess...?

**Pull request merged**